prow/cmd/webhook-server/validation.go

```diff
@@ -16,8 +16,96 @@ limitations under the License.
 
 package main
 
-import "net/http"
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
 
-func serveValidate(http.ResponseWriter, *http.Request) {
+	"github.com/sirupsen/logrus"
 
+	"k8s.io/api/admission/v1beta1"
+	apiv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/plank"
+)
+
+var agentsNotSupportingCluster = sets.NewString("jenkins")
+
+func (wa *webhookAgent) serveValidate(w http.ResponseWriter, r *http.Request) {
+    body, err := ioutil.ReadAll(r.Body) 
+    if err != nil {
+        logrus.WithError(err).Fatal("Unable to read request")
+    }
+    var prowJob v1.ProwJob
+    admissionReview := &v1beta1.AdmissionReview{}
+    err = json.Unmarshal(body, admissionReview)
+	if err != nil {
+        logrus.WithError(err).Fatal("Unable to unmarshal admission review request")
+    }
+    admissionRequest := admissionReview.Request
+    err = json.Unmarshal(admissionRequest.Object.Raw, &prowJob)
+    if err != nil {
+        logrus.WithError(err).Fatal("Unable to unmarshal admission review request")
+    }
+    var admissionResponse *v1beta1.AdmissionResponse
+    if admissionRequest.Operation == "CREATE" {
+        if err := validateProwJobClusterOnCreate(prowJob, wa.statuses); err != nil {
```

Comment:
You can simplify this by having createAdmissionResponse take an err instead of "allowed, msg, and reason."